clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -948,10 +948,12 @@ public static void atomicMoveWithFallback(Path source, Path target, boolean need
     /**
      * Flushes dirty directories to guarantee crash consistency.
      *
+     * Note: We don't fsync directories on Windows OS because otherwise it'll throw AccessDeniedException (KAFKA-13391)
+     *
      * @throws IOException if flushing the directory fails.
      */
     public static void flushDir(Path path) throws IOException {
-        if (path != null) {
+        if (path != null && !OperatingSystem.IS_WINDOWS) {
             try (FileChannel dir = FileChannel.open(path, StandardOpenOption.READ)) {
                 dir.force(true);
```

Comment:
There is a file lock on this file that causes the issue, which might be hiding another issue even on other platforms.